api/s3/src/lib.rs

```diff
@@ -140,11 +154,30 @@ where
         unimplemented!()
     }
 
+    #[tracing::instrument(
+        skip_all,
+        fields(bucket = input.bucket.as_str(), key = input.key.as_str()),
+    )]
     async fn get_object(
         &self,
-        _: dto::GetObjectRequest,
+        input: dto::GetObjectRequest,
     ) -> S3StorageResult<dto::GetObjectOutput, dto::GetObjectError> {
-        unimplemented!()
+        let path = parse_s3_path(input.bucket, input.key.clone())?;
+
+        let bytes = self
+            .0
+            .exec(GetFile { path })
+            .await
+            .map_err(|e| internal_error("GetFile operation failed", e))?
+            .ok_or(S3StorageError::Operation(
+                dto::GetObjectError::NoSuchKey(input.key),
+            ))?;
+
+        tracing::info!("OK");
+        Ok(dto::GetObjectOutput {
+            body: Some(bytes.into()),
```

Comment:
`GetObjectOutput` supports streaming body, which we definitely should use to response in a buffered manner instead of loading the whole file at once.